### PR TITLE
Palmira.benavente/reservation UI

### PR DESCRIPTION
**Pull request by PalmiraBenavente**
Added reservation Ui 

- added activity with its mvp, contract and layout
- added dialog fragment with date and time picker
- adaptation the layout fragment to view in Nexus S

image

Nexus
image
image
image

Pixel 3a
image
image
image

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.example.parkingsystem.mvp.view;
 
-import com.example.parkingsystem.SpacesParkingDialogFragment;
 import com.example.parkingsystem.databinding.FragmentDialogSpaceParkingBinding;
+import com.example.parkingsystem.fragments.SpacesParkingDialogFragment;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java:4]: are you using this import?

File: app/src/main/java/com/example/parkingsystem/activities/ParkingReservationActivity.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        this.setListener();
+    }
+
+    private void setListener() {
+        this.binding.buttonReserverParkingPickerInic.setOnClickListener(view -> presenter.onButtonPressedSelectReserver());
+        this.binding.buttonReserverParkingPickerEnd.setOnClickListener(view -> presenter.onButtonPressedSelectReserver());
+    }
+
+    public static Intent newInstance(Context context) {
+        Intent intent = new Intent(context, ParkingReservationActivity.class);
```
Comment (BQuinteros) [inline app/src/main/java/com/example/parkingsystem/activities/ParkingReservationActivity.java]: here you can return the instance without putting in a variable before

File: app/src/main/java/com/example/parkingsystem/fragments/SpacesParkingDialogFragment.java [hunk 3]
```diff
@@ -7,8 +7,8 @@
 import android.view.ViewGroup;
 import androidx.fragment.app.DialogFragment;
 import com.example.parkingsystem.databinding.FragmentDialogSpaceParkingBinding;
-import com.example.parkingsystem.mvp.contract.DialogSpaceParkingContract;
 import com.example.parkingsystem.listener.ListenerDialogFragment;
+import com.example.parkingsystem.mvp.contract.DialogSpaceParkingContract;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/fragments/SpacesParkingDialogFragment.java:11]: are you using this import?

File: app/src/main/java/com/example/parkingsystem/fragments/DateReservationDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import androidx.fragment.app.DialogFragment;
+import com.example.parkingsystem.databinding.FragmentDialogDatePickerBinding;
+
+public class DateReservationDialogFragment extends DialogFragment {
+
+    public static DateReservationDialogFragment newInstance() {
+        DateReservationDialogFragment dataReservationDialogFragment = new DateReservationDialogFragment();
```
Comment (BQuinteros) [inline app/src/main/java/com/example/parkingsystem/fragments/DateReservationDialogFragment.java]: here too you can put
`return new DateReservationDialogFragment();`

File: app/src/main/res/layout/activity_reserver_parking_lot.xml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        android:inputType="number"
+        android:text="@string/text_edit_reserver_parking_place" />
+
+    <EditText
+        android:id="@+id/edit_text_reserver_parking_code"
+        style="@style/EditTextReserverStyle"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/reserver_parking_layout_height_xbig"
+        android:layout_marginHorizontal="@dimen/reserver_parking_margin_horizontal_xbig"
+
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_reserver_parking_lot.xml]: remove this extra line pls. Does this compile?

File: app/src/main/res/layout/activity_reserver_parking_lot.xml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    <EditText
+        android:id="@+id/edit_text_reserver_parking_site"
+        style="@style/EditTextReserverStyle"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/reserver_parking_layout_height_xbig"
+        android:layout_marginHorizontal="@dimen/reserver_parking_margin_horizontal_xbig"
+        android:layout_marginTop="@dimen/reserver_parking_margin_top_big"
+        android:background="@drawable/border"
+        android:inputType="number"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_reserver_parking_lot.xml]: android:background, android:layout_marginHorizontal and android:inputType can be on the style

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -32,4 +32,29 @@
...
+    </style>
+
+    <style name="ButtonReserverStyle">
+        <item name="android:backgroundTint">@color/ming</item>
+        <item name="android:textSize">@dimen/button_text_size_big</item>
+        <item name="android:layout_marginTop">@dimen/button_main_margin_top_big</item>
+    </style>
+
+    <style name="DatePickerStyle">
+        <item name="android:background">@color/amaranth_purple</item>
```
Comment (SartiMau) [inline app/src/main/res/values/styles.xml]: an style with only one property do not make sense, please remove it

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_reserver_parking_lot.xml (hunk 6 above)
Comment (mauroLonghi) [reply app/src/main/res/layout/activity_reserver_parking_lot.xml]: inputType to

File: app/src/main/java/com/example/parkingsystem/activities/ParkingReservationActivity.java (hunk 2 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/activities/ParkingReservationActivity.java]: Done

File: app/src/main/java/com/example/parkingsystem/fragments/DateReservationDialogFragment.java (hunk 4 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/fragments/DateReservationDialogFragment.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java (hunk 1 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java:4]: Yes, I use it in the previous task to implement the dismiss of the DialogFragmnet that sets the number of spaces

File: app/src/main/java/com/example/parkingsystem/fragments/SpacesParkingDialogFragment.java (hunk 3 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/fragments/SpacesParkingDialogFragment.java:11]: Yes, I use this import for the presenter variable, at the time of creating the variable I do `DialogSpaceParkingContract.DialogFragmentPresenter`

File: app/src/main/res/layout/activity_reserver_parking_lot.xml (hunk 5 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/activity_reserver_parking_lot.xml]: Done

File: app/src/main/res/values/styles.xml (hunk 7 above)
Comment (PalmiraBenavente) [reply app/src/main/res/values/styles.xml]: Done

File: app/src/main/res/layout/activity_reserver_parking_lot.xml (hunk 6 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/activity_reserver_parking_lot.xml]: when I move the horizontal margin I get an error,` `android: layout_marginHorizontal` requires API level 26 (current min is 23)`, the solution was to change to marginStart and marginEnd, is there any other solution or recommendation?

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+    public ReservationParkingView(Activity activity) {
+        super(activity);
+    }
+
+    @Override
+    public void showDialogDataPiker() {
+        FragmentManager fragmentManager = getFragmentManager();
+        DateReservationDialogFragment dateDialogFragment = DateReservationDialogFragment.newInstance();
+        dateDialogFragment.show(fragmentManager, "dialog_fragment_date_picker");
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java]: this string should be on a const

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -23,4 +23,12 @@
...
         android:layout_marginHorizontal="@dimen/button_main_margin_horizontal_xbig"
         android:text="@string/button_main_activity_set_parking" />
+
+    <Button
+        android:id="@+id/button_main_book_parking_spaces"
+        style="@style/ButtonMainStyle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="@dimen/button_main_margin_horizontal_xbig"
+        android:text="@string/button_main_activity_reservation" />
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: text id and button id should be cohesive
if button_main_book_parking_spaces
android:text should be main_book_parking_spaces

File: app/src/main/res/layout/fragment_dialog_date_picker.xml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:background="@color/ming">
+
+    <ScrollView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        tools:ignore="MissingConstraints">
```
Comment (nicolbidone) [inline app/src/main/res/layout/fragment_dialog_date_picker.xml]: here you can reduce layouts by using scroll view as base layout

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -5,4 +5,13 @@
     <string name="toast_main_activity_select_space_parking">Amount of space selected: %1$d</string>
     <string name="text_view_dialog_fragment_set_space">Amount of spaces</string>
     <string name="button_fragment_set_parking_ok">OK</string>
+    <string name="button_main_activity_reservation">Reserve a Parking Lot</string>
+    <string name="text_activity_parking_title">BOOK YOUR PARKING LOT HERE</string>
+    <string name="button_reserver_parking_inic">Start</string>
+    <string name="button_reserver_parking_end">End</string>
+    <string name="text_edit_reserver_parking_place">    ENTER THE PLACE HERE...</string>
+    <string name="text_edit_reserver_parking_code">    ENTER THE CODE HERE...</string>
```
Comment (nicolbidone) [inline app/src/main/res/values/strings.xml]: don't force spaces please use edit text properties

File: app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java (hunk 8 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 9 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/activity_main.xml]: Done, I tried to make the change in all the names that I saw, I will keep it in mind more for the future

File: app/src/main/res/values/strings.xml (hunk 11 above)
Comment (PalmiraBenavente) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/res/layout/fragment_dialog_date_picker.xml (hunk 10 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/fragment_dialog_date_picker.xml]: Done

File: app/src/main/res/layout/fragment_dialog_date_picker.xml [hunk 12]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    tools:ignore="MissingConstraints">
```
Comment (lucianoeli) [inline app/src/main/res/layout/fragment_dialog_date_picker.xml]: shoud not use this. you are hiding warnings.
try to fix them please.

**Review by lucianoeli — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

File: app/src/main/res/layout/fragment_dialog_date_picker.xml (hunk 12 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/fragment_dialog_date_picker.xml]: Done

File: app/src/main/res/layout/activity_reserver_parking_lot.xml (hunk 6 above)
Comment (SartiMau) [reply app/src/main/res/layout/activity_reserver_parking_lot.xml]: okay, it is fine, just keep the android:layout_marginHorizontal in the view XML or add the other margins to the style

File: app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+import android.app.Activity;
+import androidx.fragment.app.FragmentManager;
+import com.example.parkingsystem.fragments.DateReservationDialogFragment;
+import com.example.parkingsystem.mvp.contract.ReservationParkingContract;
+import com.example.parkingsystem.mvp.view.base.ActivityView;
+
+public class ReservationParkingView extends ActivityView implements ReservationParkingContract.ReservationActivityView {
+
+    private final String dialogFragmentDatePicker = "dialog fragment date picker";
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java]: remember that the constants should be in upper case and with words separated by _
`private final String DIALOG_FRAGMENT_DATE_PICKER = "dialog fragment date picker";`

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java (hunk 13 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java]: Done

**Review by mauroLonghi — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**